src/target/riscv/riscv-013.c

```diff
@@ -55,13 +55,13 @@ static riscv_insn_t riscv013_read_progbuf(struct target *target, unsigned int
 		index);
 static int riscv013_invalidate_cached_progbuf(struct target *target);
 static int riscv013_execute_progbuf(struct target *target, uint32_t *cmderr);
-static void riscv013_fill_dmi_write_u64(struct target *target, char *buf, int a, uint64_t d);
-static void riscv013_fill_dmi_read_u64(struct target *target, char *buf, int a);
-static int riscv013_dmi_write_u64_bits(struct target *target);
-static void riscv013_fill_dmi_nop_u64(struct target *target, char *buf);
-static void riscv013_fill_dm_write_u64(struct target *target, char *buf, int a, uint64_t d);
-static void riscv013_fill_dm_read_u64(struct target *target, char *buf, int a);
-static void riscv013_fill_dm_nop_u64(struct target *target, char *buf);
+static void riscv013_fill_dmi_write(struct target *target, char *buf, uint64_t a, uint32_t d);
+static void riscv013_fill_dmi_read(struct target *target, char *buf, uint64_t a);
+static void riscv013_fill_dmi_nop(struct target *target, char *buf);
```

Comment:
I think these functions can be inlined into corresponding `riscv013_fill_dm_*()`.
As an alternative, `riscv013_fill_dmi_*` functions can be exposed instead of `riscv013_fill_dm_*` and the address translation can be done in `riscv_fill_dm_*` wrappers via `riscv_get_dmi_address()` (defined in `riscv.c`).
IMHO the second option is preferable, since it results in a more uniform interface (the same approach is implemented for non-batched accesses (there are `dmi_read`/`dmi_write` in `struct riscv_info`, but no `dm_read`/`dm_write`))

Reply:
I also agree that this can be improved.

As for the callbacks (function pointers) in `struct target_type riscv011_target`, `struct target_type riscv013_target` and `struct riscv_info`, I would like to clean it up and get rid of the callbacks completely in near term. Instead, I'd prefer to make direct calls to the `riscv011_*()` and `riscv013_*()` functions in the whole codebase.

For that reason, if you don't mind, I'd like to only clean up the data types and function names, but retain the structure of the function calls for now. The structural changes I would leave for a separate merge request.

Reply:
Yes, this is out of scope.